### ISSUE TITLE
FileRepository: make a more extensive tag selector

### ISSUE THREAD

**Issue by urapadmin** [arch-kiosk/arch-kiosk-office#2555]
sometimes it is necessary to see all available tags. That is not possible in the current drop down because that would pop the drop down. There are simply too many in some projects. So we need an "advanced" selector in addition (we'll keep the current one) that shows all tags, gives the chance to search and sort them in reasonable ways and maybe even give the opportunity to change or delete some (which is difficult).

As for the current selector: It might help to know that you don't have to type in a sequence of characters that a tag start with. You type in a sequence of characters that can be found somewhere in any of the tags attached to an image.

**Comment by urapadmin** [comment]
- see also and #968

**Comment by urapadmin** [comment]
- see also complaints about the way selecting images works in #2568 Tackling a tag as a whole would have been a way to address that particular use case

**Comment by luizaogs** [comment]
You'll kill me. This would be so helpful.

**Comment by urapadmin** [comment]
which part of it? Please describe your workflow and use case.

**Comment by luizaogs** [comment]
I need to be able to find the tags and the current drop down only really works if I already know what I’m looking for. After two years, I don’t remember. We use a lot of tags in MADP, why I requested this during our first season in the first place.

**Comment by urapadmin** [comment]
okay. I was planning for more here. Something that only lists all the tags is easier to achieve than bulk operations on tags, like deleting or renaming a tag.

**Comment by luizaogs** [comment]
Well those would be great eventually :) But not something I crave at the moment.

**Comment by urapadmin** [comment]
I think a sidebar with all the tags might a good idea here. I see what I can whip up...

**Comment by urapadmin** [comment]
### Kiosk 1.7.30.3
- you have a side bar button now. It fades in a side bar with additional filters.
- the first additional filter is a new tag selector: It shows all available tags
- the normal tag filter fades out while the side bar is open
- Note that the normal tag filter has an advantage over the new one: an image is shown if any of its tags **contains** the filter text. The new filter operates on whole tags, basically because you have no way to express a partial tag.
- The list of tags can be filtered itself: just type in a text and only those tags will show that contain the text
- but it can do more: You can also filter tags out by using a `-` in front of the expression.
- and it can do even more because you can chain expressions with an ` & `: `-import & 2026` show you all entries that contain `2026 `but not those with `import`. 
- the currently selected tag is always visible (just in case you wonder)
- I should add that selecting a tag in the selector does not fetch all the images immediately. It acts like all other filters: You first get the number of images that would be affected and only on an additional click the images actually download. That is a bit less intuitive with the tag selector than with the filter fields. But it burns less trees for no reason,

**Comment by urapadmin** [comment]
@luizaogs If this works (and for you) and you want it in the field, let me know. I can make you a patch for it.

**Comment by luizaogs** [comment]
This is going to be extremely helpful. 

I do have two questions: 

1) Is there a way to unselect a tag once it is selected? I've tried clicking on the tag again but that doesn't seem to do it. 

2) What is the rationale for the organization of the list? Is there one? It looks a bit jumbled at the moment. It would be helpful, for instance, if the import tags could be in order, and the rest ordered alphabetically?

<img width="225" height="315" alt="Image" src="https://github.com/user-attachments/assets/569a6a71-c1e2-4b46-a790-1bd3755223fb" />

**Comment by lbestock** [comment]
Funny it stays highlit and present even if one starts a filter search.
So I had "real llama" still with the highlighting, and I started a search "sq" and real llama sticks around.

<img width="262" height="184" alt="Image" src="https://github.com/user-attachments/assets/5eefa91a-ff68-45da-b525-56b66c88db0f" />

**Comment by urapadmin** [comment]
> Funny it stays highlit and present even if one starts a filter search. So I had "real llama" still with the highlighting, and I started a search "sq" and real llama sticks around.
> <img alt="Image" width="262" height="184" src="https://private-user-images.githubusercontent.com/49291909/539448882-5eefa91a-ff68-45da-b525-56b66c88db0f.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjkxMTk3MjEsIm5iZiI6MTc2OTExOTQyMSwicGF0aCI6Ii80OTI5MTkwOS81Mzk0NDg4ODItNWVlZmE5MWEtZmY2OC00NWRhLWI1MjUtNTZiNjZjODhkYjBmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjAxMjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwMTIyVDIyMDM0MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTZkNmZkM2ZlN2FmYmUyNGFhNzVjNGY1Yjk1NzI5NmY0YTI4MDQ0MWIxY2JmMDg5Nzg1ZWRjZTBiN2U5ZDM5NmUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.kMJTEVCVpDQlrOJq1dZdqv_Ja8zAqYBYDyta7DkMpYE">

yes, that is what I told you would happen: 
- the currently selected tag is always visible (just in case you wonder)

You wondered but didn't read :)

**Comment by urapadmin** [comment]
> This is going to be extremely helpful.
> 
> I do have two questions:
> 
>     1. Is there a way to unselect a tag once it is selected? I've tried clicking on the tag again but that doesn't seem to do it.

Currently you'd have to close the side bar and then remove the tag from the tag filter. I suspect clicking on the selected tag makes some sense to shortcut that.
> 
>     2. What is the rationale for the organization of the list? Is there one? It looks a bit jumbled at the moment. It would be helpful, for instance, if the import tags could be in order, and the rest ordered alphabetically?
> 
> 
> <img alt="Image" width="225" height="315" src="https://private-user-images.githubusercontent.com/67970402/539376426-569a6a71-c1e2-4b46-a790-1bd3755223fb.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjkxMTk4MjAsIm5iZiI6MTc2OTExOTUyMCwicGF0aCI6Ii82Nzk3MDQwMi81MzkzNzY0MjYtNTY5YTZhNzEtYzFlMi00YjQ2LWE3OTAtMWJkMzc1NTIyM2ZiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjAxMjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwMTIyVDIyMDUyMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTliMTViOTAwYjU2OGVkODA3NjVlZjBkODlhOGJhYWRkNWQ0OTdiY2QxZjZiYjA1ZGRmYTc2YjQ3NmVlM2NhM2UmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.4ME5lC7z-ETX0kWFd4dbRNPSkBfNY8XiHYbtSDio9c8">

I can order it alphabetically in general. I can not order the import tags in an order that your human eye sees (but the computer does not, it is just text).

**Comment by lbestock** [comment]
> You wondered but didn't read :)

I read, but read while zooming to a faculty meeting. That never goes well.

**Comment by urapadmin** [comment]
> > You wondered but didn't read :)
> 
> I read, but read while zooming to a faculty meeting. That never goes well.

ja, those are also my favourite tests. As if testing were something one could do with half a brain.

**Comment by urapadmin** [comment]
### Kiosk 1.7.30.4 / Kiosk 1.7.33
- alphanumeric order of the tags
- clicking on the selected tag un-selects it

**Comment by luizaogs** [comment]
It's great! Thank you. 

Should I close this or leave it open for the other stuff you said you wanted to do above? I can also open a new ticket for that if that's better (this has gotten a bit messy).

**Comment by urapadmin** [comment]
we need to close tickets. Kiosk is like a Hydra.

**Comment by luizaogs** [comment]
Ha ok. I'll open a new one then ... (which I guess is not very helpful)
